mox/impl/src/lib.rs

```diff
@@ -1,245 +1,278 @@
+        )
     }
 }
 
-struct MoxAttr {
-    name: Ident,
-    value: TokenStream,
-}
+impl<T> NodeConvertError for T where T: TryFrom<syn_rsx::Node> {}
 
-impl MoxAttr {
-    fn child(item: &MoxItem) -> Self {
-        Self { name: Ident::new("child", item.span()), value: item.to_token_stream() }
-    }
+fn node_span(node: &syn_rsx::Node) -> Span {
+    // TODO get the span for the whole node, see `https://github.com/stoically/syn-rsx/issues/4`
+    // Prioritize name's span then value's span then call site's span.
+    node.name_span()
+        .or_else(|| node.value.as_ref().map(|value| value.span()))
+        .unwrap_or_else(Span::call_site)
 }
 
-impl ToTokens for MoxAttr {
+impl ToTokens for MoxItem {
     fn to_tokens(&self, tokens: &mut TokenStream) {
-        let Self { name, value } = self;
-        let name = mangle_keywords(name);
-        tokens.extend(quote!(.#name(#value)));
+        match self {
+            MoxItem::Tag(tag) => tag.to_tokens(tokens),
+            MoxItem::Expr(expr) => expr.to_tokens(tokens),
+            MoxItem::None => (),
+        }
     }
 }
 
-fn mangle_keywords(name: &Ident) -> Cow<'_, Ident> {
-    let replacement = if name == "async" {
-        Some("async_")
-    } else if name == "for" {
-        Some("for_")
-    } else if name == "loop" {
-        Some("loop_")
-    } else if name == "type" {
-        Some("type_")
-    } else {
-        None
-    };
-
-    match replacement {
-        Some(r) => Cow::Owned(Ident::new(r, name.span())),
-        None => Cow::Borrowed(name),
-    }
-}
+impl ToTokens for MoxTag {
+    fn to_tokens(&self, tokens: &mut TokenStream) {
+        let MoxTag { name, attributes, children } = self;
 
-impl From<SnaxAttribute> for MoxAttr {
-    fn from(attr: SnaxAttribute) -> Self {
-        match attr {
-            SnaxAttribute::Simple { name, value } => {
-                let name_str = name.to_string();
-                if name_str == "_" {
-                    abort!(
-                        name.span(),
-                        "anonymous attributes are only allowed in the first position"
-                    )
-                } else {
-                    MoxAttr {
-                        name,
-                        value: match value {
-                            TokenTree::Group(g) => g.stream(),
-                            other => other.to_token_stream(),
-                        },
-                    }
-                }
-            }
+        // this needs to be nested within other token groups, must be accumulated
+        // separately from stream
+        let mut contents = quote!();
+
+        for attr in attributes {
+            attr.to_tokens(&mut contents);
         }
+
+        for child in children {
+            quote!(.child(#child)).to_tokens(&mut contents);
+        }
+
+        quote!(mox::topo::call(|| { #name() #contents .build() })).to_tokens(tokens);
```

Comment:
Could you add a TODO here referencing https://github.com/anp/moxie/issues/199?

Reply:
```suggestion
        // TODO remove `topo` dependency, see `https://github.com/anp/moxie/issues/199`
        quote!(mox::topo::call(|| { #name() #contents .build() })).to_tokens(tokens);
```